src/public.ts

```diff
@@ -0,0 +1,255 @@
+        return client;
+    }
+  
+    public async streamEvents(streamName: string,  offset: Offset = Offset.zero): Promise<PausableStream<StreamEvent>> {
+      // 1. find endpoint having the offset
+      // 2. stream events from it
+      const endpoints: StreamEndpoint[] = await this.registry.findStreamEndpoints(streamName, offset)
+      for (const endpoint of endpoints) {
+        const client = this.getStreamConsumerClient(endpoint.uri)
+        return await client.streamStateTransitions(streamName, offset)
+      }
+    throw new Error("Cannot stream data")
+  }
+}
+  
+  export class Offset {
+    public static readonly zero = new Offset("", BigInt(0), Timestamp.zero);
+  
+    public constructor(
+      public readonly id: string,
+      public readonly height: bigint,
+      public readonly timestamp: Timestamp,
+    ) {
+      assert(id.length > 0 || height == BigInt(0));
+    }
+  
+    public equals(offset: Offset): boolean {
+      return this.id == offset.id;
+    }
+  
+    public sameHeight(offset: Offset): boolean {
+      return this.height == offset.height;
+    }
+  
+    public canSucceed(offset: Offset): boolean {
+      return this.height - BigInt(1) == offset.height
+        && this.timestamp.greaterThan(offset.timestamp);
+    }
+  
+    public canPrecede(offset: Offset): boolean {
+      if (this.equals(Offset.zero))
+        return true;
+  
+      return this.height + BigInt(1) == offset.height
+        && this.timestamp.lessThan(offset.timestamp);
+    }
+  
+    public dump(): string {
+      return `${this.height}-${this.id}@(${this.timestamp.dump()})`
+    }
+  }
+  
+  
+  export class StreamEvent {
+    public constructor(
+      public readonly offset: Offset,
+      public readonly payload: Uint8Array,
+      public readonly timestamp: Timestamp,
+      public readonly undo: boolean,
+    ) {
+    }
+  }
+  
+  export interface StreamClientOptions {
+    registry?: StreamRegistry;
+  }
+
+  export interface StreamRegistryOptions {
+    retryPolicy: {
+        retryCount: number;
+        waitInMs: number;
+     }
+  }
+
+  export const DefaultRegistryClientOptions: StreamRegistryOptions = {
+    retryPolicy: {
+        retryCount: 10,
+        waitInMs: 300
+    }
+  }
+  
+  //Stream Registry (get stream and get streams?)
+  export interface StreamRegistry {
+    findStreamEndpoints(streamName: string, offset: Offset): Promise<StreamEndpoint[]>;
+    findStreams(streamFilter: StreamFilter): Promise<Stream[]> 
+    getStream(streamName: string): Promise<Stream> 
+  }
+  
+  export class RemoteStreamRegistry implements StreamRegistry, StreamDiscovery {
+    public constructor(private readonly endpoint: string = "https://stream-api.cluster.amur-dev.proxima.one", public readonly options: StreamRegistryOptions = DefaultRegistryClientOptions) { }
+  
+    public async findStreamEndpoints(streamName: string, from: Offset, to?: Offset): Promise<StreamEndpoint[]> {
```

Comment:
I think it should be [`FindOffsetFilter`](https://github.com/proxima-one/stream-registry-api/blob/db52a30fafb24c9a393de61efc99934277890902/src/model/stream.ts#L34) instead of `Offset` because providing an `""` as an id makes invalid offset

Reply:
Added the filter in this iterattion iteration.